Modules/socketmodule.c

```diff
@@ -2583,7 +2605,7 @@ sock_settimeout(PySocketSockObject *s, PyObject *arg)
         return NULL;
 
     s->sock_timeout = timeout;
-    if (internal_setblocking(s, timeout < 0) == -1) {
+    if (internal_setblocking(s, timeout != 0) == -1) {
```

Comment:
Hum, wait, what is this change? Timeouts > 0 now set the socket is blocking mode?

Would you mind to document this subtle behaviour change in settimeout() documentation?

Reply:
Yes.  I'm still working on it.  Turns out sendfile currently tests test sendfile on non-blocking sockets, but at the same time the documentation says that it's not supported for non-blocking sockets at all.  It's a mess.